python/ray/rllib/dqn/dqn_evaluator.py

```diff
@@ -59,14 +59,28 @@ def update_target(self):
 
     def sample(self):
         obs, actions, rewards, new_obs, dones = [], [], [], [], []
-        for _ in range(self.config["sample_batch_size"]):
+        for _ in range(
+                self.config["sample_batch_size"] + self.config["n_step"] - 1):
```

Comment:
why is this the sum of these two values?

Reply:
If you have N step learning, your batch size must be at least B+N steps in order to produce B steps after post-processing.

Reply:
Shouldn't this actually be `max(B, N)`?

Also, this should probably just be refactored into using `process_rollout` (can be done later)

Reply:
Well if you want to produce a batch of size B you need B+N steps (since the edges get cut off).

I do agree this stuff could use some refactoring, probably will have to happen as part of the external data API anyways.